packages/cli/src/models/local/LocalBaseController.js

```diff
@@ -110,6 +115,57 @@ export default class LocalBaseController {
     return this.hasTypeIdentifier(contractDataPath, "t_function_baredelegatecall_nonpayable$__$returns$_t_bool_$")
   }
 
+  getUninitializedBaseContracts(contractDataPath) {
+    if (!fs.exists(contractDataPath)) return []
+    const contractJson = fs.parseJson(contractDataPath)
+    // Check whether the contract has base contracts
+    const baseContracts = contractJson.ast.nodes.find(n => n.name === contractJson.contractName).baseContracts
+    if (baseContracts.length == 0) {
+      return []
+    }
+    // Make a dict of base contracts that have "initialize" function
+    const baseContractsWithInitialize = {}
+    for (const baseContract of baseContracts) {
+      const baseContractName = baseContract.baseName.name
+      const baseContractPath = Contracts.getLocalPath(baseContractName)
+      const baseContractAbi = fs.parseJson(baseContractPath).abi
+      if (baseContractAbi.find(fn => fn.type === "function" && fn.name === "initialize")) {
+        baseContractsWithInitialize[baseContractName] = false
+      }
+    }
+    // Find "initialize" function in the AST
+    const contractDefinition = contractJson.ast.nodes
+      .find(n => n.nodeType === "ContractDefinition" && n.name === contractJson.contractName)
+    const initializeFunction = contractDefinition.nodes
+      .find(n => n.nodeType === "FunctionDefinition" && n.name === "initialize")
+    if (initializeFunction === undefined) {
+      return []
+    }
+    const initializedContracts = {}
+    // Update map with each call of "initialize" function of the base contract
+    for (const statement of initializeFunction.body.statements) {
+      if (statement.nodeType !== "ExpressionStatement") {
+        continue
+      }
+      if (statement.expression.nodeType !== "FunctionCall") {
+        continue
+      }
+      if (statement.expression.expression.memberName !== "initialize") {
+        continue
+      }
+      const baseContractName = statement.expression.expression.expression.name
+      initializedContracts[baseContractName] = true
```

Comment:
Minor style comment: could we rephrase these checks as the following?
```js
if (statement.nodeType === "ExpressionStatement" && statement.expression.nodeType === "FunctionCall" && statement.expression.expression.memberName === "initialize") {
  const baseContractName = statement.expression.expression.expression.name
  initializedContracts[baseContractName] = true
}
```

Reply:
Sure!